### DISCUSSION TITLE
Custom message generation using ros2cs

### DISCUSSION THREAD

**Discussion by kinrep71** [noelex/rclnet#5] [Q&A]
Hi all,

I would like to generate a custom message using ros2cs. Is this source generator supposed to be used for these purposes? 
![image](https://user-images.githubusercontent.com/49516152/233299817-8c208e4d-214f-4019-b862-d19eccd238cf.png)
Is the description of the custom message generation workflow available somewhere? 

Thank You in advance :)

**Comment by noelex** [comment]
I would recommend you to install `Rcl.NET` and `ros2cs` utility via NuGet:
```
dotnet tool install -g ros2cs
```
To generate messages, prepare a `ros2cs.spec` and run `ros2cs /path/to/your/ros2cs.spec`.

See [here](https://github.com/noelex/rclnet#installing) for detailed instructions.
